test_main.py

```diff
@@ -37,3 +42,57 @@ def test_import():
     finally:
         os.environ.clear()
         os.environ.update(orig_env)
+
+
+def test_url_caching_with_cloud_function(empty_deployment):
+    """kcidb cache client workflow test"""
+    assert empty_deployment is None
+
+    pub_client = kcidb.mq.URLListPublisher(
+        project_id=os.environ["GCP_PROJECT"],
+        topic_name=os.environ["KCIDB_UPDATED_URLS_TOPIC"]
+    )
+
+    # Submit messages with different URLs
+    urls_messages = [
+        ["https://github.com/kernelci/kcidb/blob/main/setup.py"],
+        ["https://github.com/kernelci/kcidb/blob/main/requirements.txt",
+         "https://github.com/kernelci/kcidb/blob/main/README.md"]
+    ]
+
+    for urls in urls_messages:
+        pub_client.publish(urls)
+
+    # Retry checking URLs in the cache for a minute
+    retry_interval = 5  # seconds
+    max_retries = 12  # 60 seconds / 5 seconds
+
+    for urls in urls_messages:
+        for url in urls:
+            for _ in range(max_retries):
+                if check_url_in_cache(url):
+                    break
+                time.sleep(retry_interval)
+            else:
+                raise AssertionError(f"URL '{url}' not found in the cache")
+
+
+def check_url_in_cache(url):
+    """URL check if its coming from storage or not"""
```

Comment:
Please rephrase using the imperative mood.

Reply:
okay